src/services/useWms/index.ts

```diff
@@ -0,0 +1,132 @@
+import {
+  BoundingBox,
+  GetCapabilitiesResponse,
+  Layer,
+  WMSProvider,
+} from '@deltares/fews-wms-requests'
+import { MaybeRefOrGetter, ref, Ref, toValue, watchEffect } from 'vue'
+// @ts-ignore
+import { toWgs84 } from '@turf/projection'
+// @ts-ignore
+import { point } from '@turf/helpers'
+import { LngLatBounds } from 'mapbox-gl'
+import { GetLegendGraphicResponse } from '@deltares/fews-wms-requests/src/response/getLegendGraphicResponse.ts'
+
+export interface UseWmsReturn {
+  legendGraphic: Ref<GetLegendGraphicResponse | undefined>
+  times: Ref<Date[] | undefined>
+}
+
+export function useWmsLayer(
+  baseUrl: string,
+  layerName: MaybeRefOrGetter,
+): UseWmsReturn {
+  const legendGraphic = ref<GetLegendGraphicResponse>()
+  let wmsUrl = `${baseUrl.toString()}/wms`
+  const wmsProvider = new WMSProvider(wmsUrl)
+  const times = ref<Date[]>()
+
+  async function loadTimes(): Promise<void> {
+    try {
+      let layers = toValue(layerName)
+      const capabilities = await wmsProvider.getCapabilities({
+        layers: layers,
+        importFromExternalDataSource: false,
+        onlyHeaders: false,
+        forecastCount: 1,
+      })
+      let valueDates: Date[]
+      let selectedLayer: Layer
+      if (capabilities.layers.length > 0) {
+        selectedLayer = capabilities.layers[0]
+        capabilities.layers.forEach((l) => {
+          if (l.name === layers) {
+            selectedLayer = l
+          }
+        })
+        if (selectedLayer.times) {
+          const dates = selectedLayer.times.map((time) => {
+            return new Date(time)
+          })
+          let firstValueDate = dates[0]
+          let lastValueDate = dates[dates.length - 1]
+          if (selectedLayer.firstValueTime) {
+            firstValueDate = new Date(selectedLayer.firstValueTime)
+          }
+          if (selectedLayer.lastValueTime) {
+            lastValueDate = new Date(selectedLayer.lastValueTime)
+          }
+          valueDates = dates.filter(
+            (d) => d >= firstValueDate && d <= lastValueDate,
+          )
+          if (
+            selectedLayer.keywordList &&
+            selectedLayer.keywordList[0].forecastTime
+          ) {
+            //this.externalForecast = new Date(selectedLayer.keywordList[0].forecastTime)
+          }
+        } else {
+          valueDates = []
+        }
+        times.value = valueDates
+      } else {
+        times.value = []
+      }
+    } catch (error) {
+      console.log(error)
+    }
+  }
+
+  async function loadCapabilities(): Promise<void> {
+    try {
+      legendGraphic.value = await wmsProvider.getLegendGraphic({
+        layers: toValue(layerName),
+      })
```

Comment:
Check is `layerName` is not an empty string. This will prevent an invalid request.